contracts/utils/OpsReady.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {
+    SafeERC20,
+    IERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IOps {
+    function gelato() external view returns (address payable);
+    
+    function createTask(
+        address _execAddress,
+        bytes4 _execSelector,
+        address _resolverAddress,
+        bytes calldata _resolverData
+    ) external returns (bytes32 task);
+
+    function cancelTask(bytes32 _taskId) external;
+
+    function getFeeDetails() external view returns (uint256, address);
+}
+
+abstract contract OpsReady {
+    address public ops;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    modifier onlyOps() {
+        require(msg.sender == ops, "OpsReady: onlyOps");
+        _;
+    }
+
+    function gelato() public view returns (address payable) {
+        return IOps(ops).gelato();
+    }
+
+    function _transfer(uint256 _amount, address _paymentToken) internal {
```

Comment:
Not sure if you implemented this or got it from Gelato. Referring to a comment I made above about checking that `_transfer()` was successful, I see this will handle it correctly here

Reply:
Yeah gelato